### PR TITLE
Update adding tracks documentation

### PR DESCRIPTION
**Pull request by uparamonau-es**
Updating Rally documentation to fix inconsistencies discovered during onboarding.

### CODE REVIEW

**Comment by uparamonau-es** [comment]
All the listed JSONs/code/command lines have been verified with ES 6.5.3 and 7.14.1 up to Advanced Topics section which seems not so type/mapping-specific as the first part.

File: docs/adding_tracks.rst [hunk 1]
```diff
@@ -101,34 +101,40 @@ Then store the following mapping file as ``index.json`` in the tutorial director
...
+          "population": {
+            "type": "long"
           }
         }
       }
     }
 
 .. note::
-   This tutorial assumes that you want to benchmark a version of Elasticsearch prior to 7.0.0. If you want to benchmark Elasticsearch 7.0.0 or later you need to remove the mapping type above.
+   This tutorial assumes that you want to benchmark a version of Elasticsearch 7.0.0 or later. If you want to benchmark Elasticsearch prior to 7.0.0 you need to add the mapping type above so ''index.json'' will looks like::
```
Comment (pquentin) [inline docs/adding_tracks.rst:129]:
```suggestion
   This tutorial assumes that you want to benchmark a version of Elasticsearch 7.0.0 or later. If you want to benchmark Elasticsearch prior to 7.0.0 you need to add the mapping type above so ``index.json`` will look like::
```

File: docs/adding_tracks.rst [hunk 2]
```diff
@@ -212,7 +217,18 @@ Finally, store the track as ``track.json`` in the tutorial directory::
 The numbers under the ``documents`` property are needed to verify integrity and provide progress reports. Determine the correct document count with ``wc -l documents.json``. For the size in bytes, use ``stat -f %z documents.json`` on macOS and ``stat -c %s documents.json`` on GNU/Linux.
 
 .. note::
-   This tutorial assumes that you want to benchmark a version of Elasticsearch prior to 7.0.0. If you want to benchmark Elasticsearch 7.0.0 or later you need to remove the ``types`` property above.
+   This tutorial assumes that you want to benchmark a version of Elasticsearch 7.0.0 or later. If you want to benchmark Elasticsearch prior to 7.0.0 you need to add the ``types`` property above so ''track.json'' will looks like::
```
Comment (pquentin) [inline docs/adding_tracks.rst:220]:
```suggestion
   This tutorial assumes that you want to benchmark a version of Elasticsearch 7.0.0 or later. If you want to benchmark Elasticsearch prior to 7.0.0 you need to add the ``types`` property above so ``track.json`` will look like::
```

File: docs/adding_tracks.rst [hunk 3]
```diff
@@ -267,7 +283,12 @@ You can also show details about your track with ``esrally info --track-path=~/ra
     5. force-merge
     6. query-match-all (8 clients)
 
-Congratulations, you have created your first track! You can test it with ``esrally race --distribution-version=7.0.0 --track-path=~/rally-tracks/tutorial``.
+Congratulations, you have created your first track! You can test it with ``esrally race --distribution-version=7.14.1 --track-path=~/rally-tracks/tutorial``.
+
+.. note::
+
+    To test the track with Elasticsearch prior 7.0.0 you need to update ''index.json'' and ''track.json'' as specified in notes above and then execute ``esrally race --distribution-version=6.5.3 --track-path=~/rally-tracks/tutorial``.
```
Comment (pquentin) [inline docs/adding_tracks.rst:290]:
```suggestion
    To test the track with Elasticsearch prior to 7.0.0 you need to update ``index.json`` and ``track.json`` as specified in notes above and then execute ``esrally race --distribution-version=6.5.3 --track-path=~/rally-tracks/tutorial``.
```

File: docs/adding_tracks.rst [hunk 4]
```diff
@@ -369,6 +389,10 @@ To specify different workloads in the same track you can use so-called challenge
 
     If you define multiple challenges, Rally runs the challenge where ``default`` is set to ``true``. If you want to run a different challenge, provide the command line option ``--challenge=YOUR_CHALLENGE_NAME``.
 
+.. note::
+
+    To use the track with Elasticsearch prior 7.0.0 you need to update ''index.json'' and ''track.json'' with index and mapping types accordingly as specified in notes above.
```
Comment (pquentin) [inline docs/adding_tracks.rst:394]:
```suggestion
    To use the track with Elasticsearch prior to 7.0.0 you need to update ``index.json`` and ``track.json`` with index and mapping types accordingly as specified in notes above.
```

File: docs/adding_tracks.rst [hunk 5]
```diff
@@ -866,6 +887,10 @@ We also derive an appropriate index and document type from the track's index def
       "type": "docs"
     }
 
+.. note::
+
+    Please remember about index and mapping types usage in ''index.json'' and ''track.json'' in Elasticsearch prior 7.0.0 as specified in notes above.
```
Comment (pquentin) [inline docs/adding_tracks.rst:892]:
```suggestion
    Please remember about index and mapping types usage in ``index.json`` and ``track.json`` in Elasticsearch prior to 7.0.0 as specified in notes above.
```

**Review by pquentin — CHANGES_REQUESTED** [review]
Thanks! I like this a lot because:

 1. 7.x (and soon 8.x) become the normal cases instead of the exceptional one
 2. When we drop support for 6.x, this page will be easy to update (just remove a bunch of stuff)

I left a few suggestions about typography mostly, you should be able to commit them directly from the GitHub interface.

**Review by pquentin — APPROVED** [review]

**Pull request merged**